sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -264,10 +257,16 @@ internal bool ShouldUseDefaultMessageLoggingPolicy()
             && _enableMessageLogging == null
             && _enableMessageContentLogging == null
             && _loggerFactory == null
-            && _allowedHeaderNamesHasChanged != true // false or null
-            && _allowedQueryParametersHasChanged != true; // false or null
+            && HeaderListIsDefault
+            && QueryParameterListIsDefault;
     }
 
+    private bool HeaderListIsDefault =>
```

Comment:
I personally like to break complicated boolean statements into smaller pieces that are easier to reason about.  I find it is easier to validate correctness at a glance for myself when authoring code, and I feel it also simplifies the code so that people adding a feature to the library or doing other types of maintenance in the library will have a better chance of understanding what this type is about and how it works.